contracts/child/ChildToken.sol

```diff
@@ -36,10 +36,10 @@ contract ChildToken is Ownable {
     uint256 output1
   );
 
-  function deposit(address user, uint256 amountOrTokenId) public onlyOwner;
+  function deposit(address user, uint256 amountOrTokenId) public;
 
   function withdraw(uint256 amountOrTokenId) public;
 
-  function setParent(address parent) public isParentOwner;
+  function setParent(address _parent) public;
```

Comment:
Since this contract `is Ownable`, the `parentOwner` variable can be removed.
What does the `parent` of the token contract mean? and why did you remove the `isParentOwner` modifier from `setParent`?

Reply:
because abstract functions can't have modifiers.
`parent ` is for before transfer contract and `parentOwner ` is its owner.

Reply:
As discussed, `parentOwner` design is a little twisted, but ok for now.